code/modules/projectiles/projectile.dm

```diff
@@ -631,4 +631,64 @@ var/list/impact_master = list()
 
 /obj/item/projectile/attack_hand(mob/user)
 	if(timestopped)
-		..()
\ No newline at end of file
+		..()
+
+/obj/item/projectile/friendlyCheck
+	invisibility = 101
+	rotate = 0
+	damage = 0
+	nodamage = 1
+	var/atom/impact = null
+
+/obj/item/projectile/friendlyCheck/process()
+	OnFired()
+	while(!impact && loc && (kill_count > 0))
+		if(dist_x > dist_y)
+			bresenham_step(dist_x,dist_y,dx,dy)
+		else
+			bresenham_step(dist_y,dist_x,dy,dx)
+	return impact
+
+/obj/item/projectile/friendlyCheck/bresenham_step(var/distA, var/distB, var/dA, var/dB)
+	kill_count--
+	total_steps++
+	if(error < 0)
+		var/atom/step = get_step(src, dB)
+		if(!step)
+			bullet_die()
+		src.Move(step)
+		error += distA
+		bump_original_check()
+	else
+		var/atom/step = get_step(src, dA)
+		if(!step)
+			bullet_die()
+		src.Move(step)
+		error -= distB
+		dir = dA
+		if(error < 0)
+			dir = dA + dB
+		bump_original_check()
+
+/obj/item/projectile/friendlyCheck/Bump(atom/A as mob|obj|turf|area)
```

Comment:
It looks like copypaste with the vehicle and mecha checks, and you should always make repeated code a proc if it has to work the same every time.
Unless you can think of a reason why the way this gets mobs shouldn't change in the future with new systems, it should be a proc.

Also the as mob|obj|turf|area is a dead giveaway because it's totally unnecessary and no new code should be using it in this context.